perf_dashboard/helpers/outlier_detection.py

```diff
@@ -0,0 +1,79 @@
+# Copyright Istio Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+def find_outliers(pattern_data):
+    perf_val = []
+    for data in pattern_data:
+        if data[3] != 'null':
+            perf_val.append((data[3], data[4], data[5]))
+
+    perf_val.sort(key=lambda x: x[0])
+    lower_inner_fence, upper_inner_fence, lower_outer_fence, upper_outer_fence = calculate_fences(perf_val)
+    mild_outliers = []
+    extreme_outliers = []
+    outliers = []
+    for val in perf_val:
+        if ((val[0] > upper_inner_fence and val[0] < upper_outer_fence) or (val[0] < lower_inner_fence and val[0] > lower_outer_fence)):
+            mild_outliers.append((val[0], val[1], val[2]))
+        elif val[0] > upper_outer_fence or val[0] < lower_outer_fence:
+            extreme_outliers.append((val[0], val[1], val[2]))
+    outliers = mild_outliers + extreme_outliers
+    return outliers
+
+
+def calculate_fences(perf_val):
+    Q1, Q3, IQR = cacluate_interquartile_range(perf_val)
+    lower_inner_fence = Q1 - 1.5 * IQR
+    upper_inner_fence = Q3 + 1.5 * IQR
+    lower_outer_fence = Q1 - 3 * IQR
+    upper_outer_fence = Q3 + 3 * IQR
+    return lower_inner_fence, upper_inner_fence, lower_outer_fence, upper_outer_fence
+
+
+def cacluate_interquartile_range(perf_val):
```

Comment:
are there any python std library to do this sort of thing for us?

Reply:
maybe have, but I think we may want more customized stuff later.